lib/NmfTextTagger.jsm

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+this.NmfTextTagger = class NmfTextTagger {
+  constructor(model, tokenizer) {
+    this.model = model;
+    this.tokenizer = tokenizer;
+  }
+
+  /**
+   * Determines if the text belongs to class according to binary naive Bayes
```

Comment:
I think there is a typo in this comment, it mentions "naive bayes" here. Pretty sure it should be talking about "Nonnegative Matrix Factorization"